(fix) O3-3709 - ward app - handle pagination

chibongho opened the pull request:
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://om.rs/o3ui).
- [ ] My work includes tests or is validated by existing tests.

## Summary
<!-- Please describe what problems your PR addresses. -->
NOTE: This PR is *not* ready to merge in yet, as it depends on this corresponding [PR](https://github.com/openmrs/openmrs-esm-core/pull/1123#issuecomment-2310980464) in core. It's also failing a few test cases I didn't touch. I couldn't figure it out, but I suspect it's because I'm doing `yarn link` to my core library locally.

That said, it should be ready for review.

This PR goes through places in the ward app that uses `openmrsFetch` and replaces the `useSWR` or `useSWRImmutable` functions where appropriate. 

Here's a list of changes that were made / reviewed:
```

# done
src/hooks/useObs.ts
src/hooks/useConcept.ts
src/ward-workspace/ward-patient-notes/notes.resource.ts
src/ward-patient-card/row-elements/ward-patient-obs.resource.ts
src/hooks/useInpatientRequest.ts
src/hooks/useInpatientAdmission.ts
src/hooks/useLocations.ts # fhir endpoint
src/hooks/useBeds.ts

# no change needed, post request
src/ward.resource.ts

# no change needed, hook returns single result:
src/hooks/useRestPatient.ts
src/hooks/useMostRecentObs.ts
src/hooks/useLocation.ts
src/hooks/useAdmissionLocation.ts

# no changed needed, no pagination:
src/hooks/useEmrConfiguration.ts

# no changes needed, test file:
src/ward-workspace/admit-patient-form-workspace/admit-patient-form.test.tsx

```

## Screenshots
<!-- Required if you are making UI changes. -->
cursory test of various scenarios that loads data from server:
https://github.com/user-attachments/assets/64a68ef0-48c8-4047-9c6b-0b797acf8e24



## Related Issue
<!-- Paste the link to the Jira ticket here if one exists. -->
<!-- https://issues.openmrs.org/browse/O3- -->

## Other
<!-- Anything not covered above -->
cc: @kb019

mogoodrich commented on packages/esm-appointments-app/translations/en.json:
```diff
@@ -62,6 +62,7 @@
   "createNewAppointment": "Create new appointment",
   "date": "Date",
   "date&Time": "Date & time",
+  "dateAppointmentIssuedCannotBeAfterAppointmentDate": "Date appointment issued cannot be after the appointment date",
```
This is likely okay, looks like it's coming from a previous change where the hook to update translations must not have been run.

mogoodrich commented on package.json:
```diff
@@ -84,5 +84,8 @@
-  "packageManager": "yarn@4.2.2"
+  "packageManager": "yarn@4.2.2",
+  "resolutions": {
+    "@openmrs/esm-react-utils": "portal:/home/cbho/workspace/openmrs-esm-core/packages/framework/esm-react-utils"
```
hardcoded to your workspace?

chibongho replied:
Yeah... I can't have a correct version of this until the corresponding core PR is merged (hence it's still in draft mode). I'll change after that other PR is merged.

mogoodrich reviewed (changes requested):
Not saying I 100% follow, but it looks like this is mainly updating the code to use the new pagination hooks you added.  Sounds great!  Note my one comment where you appear to be harding in a local path.

CynthiaKamau commented on packages/esm-ward-app/src/ward-view/ward-view.resource.ts.orig:
```diff
@@ -0,0 +1,133 @@
+    const { patients } = bedLayout;
+    patients.map((patient) => {
+      const patientAdmittedWithBed = inpatientAdmissionsByPatientUuid.get(patient.uuid);
+<<<<<<< HEAD
```
Looks like a rebase is still ongoing

chibongho commented:
Ok, this is out of draft mode now. Updated yarn.lock to remove local dependency onto core, and added a mock in ward-view.test.tsx to get the tests to pass.
